### ISSUE TITLE
Currently cleaning zones not displayed

### ISSUE THREAD

**Issue by ottelo9** [TheLastProject/lovelace-valetudo-map-card#78]
I cannot see any of my configured cleaning zones (two zones configured in valetudo) if I start zoned cleaning. I can only see the map, the icons, and the controls (and of course the black lines/routes).

Second question is: I add custom buttons to clean my zones, but cannot set any name. I have few zones (kitchen, floor, living room) and cannot find any suitable icon for it.

I use Xiaomi Roborock S5(0) with newest Valetudo 0.6.1, mqtt and the newest HA+Hass.io (on Rpi3).

My map card:
```
type: 'custom:valetudo-map-card'
entity: sensor.xiaomi_map
vacuum_entity: vacuum.staubsauger
title: Staubsauger Karte
rotate: 0
map_scale: 3
crop:
  top: 0
  bottom: 0
  left: 0
  right: 0
min_height: 0
custom_buttons:
  - service: script.staubsauger_zone_flur
```

**Comment by TheLastProject** [comment]
For your second question, I'll just implement https://github.com/TheLastProject/lovelace-valetudo-map-card/issues/65 as a string next to the button as I originally intended.

The zones... are the available over MQTT? I thought they weren't and that is why I created https://github.com/Hypfer/Valetudo/pull/507.

**Comment by ottelo9** [comment]
Ah ok its actually broken? Because you have zone options in

https://github.com/TheLastProject/lovelace-valetudo-map-card#options

**Comment by TheLastProject** [comment]
Oh, it may be. I never use zoned cleanup. I understand you better now, I'll see if I can reproduce the issue.

**Comment by ottelo9** [comment]
For your questions about zones over mqtt. In the mqtt settings in valetudo there is an options box named "provide map data". But I think this is for the live map (png file?!).

If I check my roborock device under mqtt integration section (Home Assistant) I can see following Subscribed topics:

valetudo/Staubsauger/status
valetudo/Staubsauger/attributes
valetudo/Staubsauger/state

and MQTT discovery data:
homeassistant/vacuum/valetudo_Staubsauger/config

under "valetudo/Staubsauger/attributes" I get message like this:

Received 17:14:37
QoS: 0
Payload: mainBrush: 144.8
sideBrush: 44.8
filter: 128.9
sensor: 8.9
cleanTime: 154.8
cleanArea: 8681.6
cleanCount: 593
lastCleanedArea: 4
lastCleaningTime: 3
last_run_stats:
  startTime: 1609257446000
  endTime: 1609257486000
  duration: 40
  area: 0.3
  errorCode: 0
  errorDescription: No error
  finishedFlag: false
state: cleaning
valetudo_state: cleaning
zoneStatus:
  - Flur

zoneStatus => my actual choosen zone (room to clean) to clean.

**Comment by ottelo9** [comment]
Am I right That your commit #507 on Valetudo is already included in 0.6.1, so that I can use it with these entry:

```
sensor:
  - platform: mqtt
    state_topic: "valetudo/rockrobo/state"
    json_attributes_topic: "valetudo/rockrobo/zones"
    name: xiaomi_map_zones
    value_template: 'OK'
    scan_interval: 5
```

**Comment by TheLastProject** [comment]
No, it is not included. You can see that because it is closed, not merged. But need something similar to be able to implement it I think

**Comment by ottelo9** [comment]
Here are few screenshots showing my problem.

Map in HA showing only the map with forbidden zone
![2021-01-02 16_18_46-Window](https://user-images.githubusercontent.com/33122175/103460438-74417c00-4d16-11eb-8cee-271f9120a430.png)

Map in Valetudo showing my cleaning zone:
![2021-01-02 16_18_24-Window](https://user-images.githubusercontent.com/33122175/103460439-74da1280-4d16-11eb-96dd-8227e45968b2.png)

**Comment by TheLastProject** [comment]
Well, my point is, if Valetudo doesn't show that zone info in MQTT somehow I can't show it either. I've not really had time to double check if it updates, which is kinda hard to do too because looking at the Valetudo MQTT entity in Home Assistant to check what is exactly in their while cleaning kinda locks up Home Assistant's UI

**Comment by ottelo9** [comment]
Hypfer released a new Valetudo version (Valetudo 2021.01.0b0) with completely rewritten mqtt.

**Comment by TheLastProject** [comment]
Someone else helped me debug this recently and this should be fixed in b32c9f41b686af0fac46c7e64f51a4d562c8fc13. Please reopen if it's still not fixed.
